src/Refitter.Core/RefitGenerator.cs

```diff
@@ -27,13 +28,94 @@ private RefitGenerator(RefitGeneratorSettings settings, OpenApiDocument document
     /// <returns>A new instance of the <see cref="RefitGenerator"/> class.</returns>
     public static async Task<RefitGenerator> CreateAsync(RefitGeneratorSettings settings)
     {
+        OpenApiDocument document;
         if (IsHttp(settings.OpenApiPath) && IsYaml(settings.OpenApiPath))
-            return new RefitGenerator(settings, await OpenApiYamlDocument.FromUrlAsync(settings.OpenApiPath));
-        if (IsHttp(settings.OpenApiPath))
-            return new RefitGenerator(settings, await OpenApiDocument.FromUrlAsync(settings.OpenApiPath));
-        if (IsYaml(settings.OpenApiPath))
-            return new RefitGenerator(settings, await OpenApiYamlDocument.FromFileAsync(settings.OpenApiPath));
-        return new RefitGenerator(settings, await OpenApiDocument.FromFileAsync(settings.OpenApiPath));
+        {
+            document = await OpenApiYamlDocument.FromUrlAsync(settings.OpenApiPath);
+        }
+        else if (IsHttp(settings.OpenApiPath))
+        {
+            document = await OpenApiDocument.FromUrlAsync(settings.OpenApiPath);
+        }
+        else if (IsYaml(settings.OpenApiPath))
+        {
+            document = await OpenApiYamlDocument.FromFileAsync(settings.OpenApiPath);
+        }
+        else
+        {
+            document = await OpenApiDocument.FromFileAsync(settings.OpenApiPath);
+        }
+
+        ProcessTagFilters(document, settings.IncludeTags);
+        ProcessPathFilters(document, settings.IncludePathMatches);
+
+        return new RefitGenerator(settings, document);
+    }
+
+    private static void ProcessTagFilters(OpenApiDocument document,
```

Comment:
I really like this approach. It seems so much more simple to just trim the OpenAPI document down instead of  checking every single endpoint upon generation